asset-registry/src/lib.rs

```diff
@@ -0,0 +1,275 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::traits::EnsureOrigin;
+use frame_support::weights::constants::WEIGHT_PER_SECOND;
+use frame_support::{log, pallet_prelude::*};
+use frame_system::pallet_prelude::*;
+pub use module::*;
+use orml_traits::asset_registry::{FixedConversionRateProvider, WeightToFeeConverter};
+use scale_info::TypeInfo;
+use sp_runtime::{traits::Member, DispatchResult};
+use sp_std::prelude::*;
+use xcm::latest::prelude::*;
+use xcm_builder::TakeRevenue;
+use xcm_executor::{traits::WeightTrader, Assets};
+
+pub trait AssetProcessor<AssetId, Metadata> {
+	fn process_asset(id: Option<AssetId>, asset_metadata: &Metadata) -> Result<(AssetId, Metadata), DispatchError>;
+}
+
+#[derive(scale_info::TypeInfo, Encode, Decode, Clone, Eq, PartialEq, Debug)]
+pub struct AssetMetadata<T: Parameter + Member + TypeInfo> {
+	pub decimals: u32,
+	pub name: Vec<u8>,
+	pub symbol: Vec<u8>,
+	pub location: Option<MultiLocation>,
```

Comment:
Good question, I'm not sure.. I think `MultiLocation` is easiest to deal with in the parachains, but for off-chain interactions (e.g. reading from js) a versionedLocation might make sense. Anyone else has any thoughts on this?

Reply:
Currently `MultiLocation` is referring to the latest version, which will change in future and causing expected issue.
So instead of using the latest, it should use v2 (and move to v3 later if needed) so ensure it doesn't change unexpectedly.

`VersionedMultiLocation` will also avoid this issue here